Cython/Compiler/Nodes.py

```diff
@@ -885,6 +887,15 @@ def declared_name(self):
     def name_cstring(self):
         return self.name.as_c_string_literal()
 
+    @property
+    def type_set_manually(self):
+        if self._type_set_manually:
+            return True
+        return bool(getattr(self.base_type, "name", False))
+    @type_set_manually.setter
+    def type_set_manually(self, value):
+        self._type_set_manually = value
```

Comment:
This feels like a work-around for some other place in the code that isn't smart enough to determine a type correctly.
Do you see a way to do this differently, without such a flag?

Reply:
I've replaced it with a flag stored on `FuncDefNode` which is probably the right place to store this information.